libtenzir/builtins/formats/lines.cpp

```diff
@@ -104,8 +105,49 @@ class lines_parser final : public plugin_parser {
   parser_args args_;
 };
 
-class plugin final : public virtual parser_plugin<lines_parser> {
+// To get better diagnostics in `write`, i.e.:
+//    "'xsv' does not support heterogeneous outputs..."
+// -> "'lines' does not..."
+// Otherwise, we could just use the `ssv_plugin` directly
+class lines_printer final : public plugin_printer {
 public:
+  lines_printer() = default;
+
+  lines_printer(std::unique_ptr<plugin_printer> inner)
+    : inner_(std::move(inner)) {
+  }
+
+  auto name() const -> std::string override {
+    return "lines";
+  }
+
+  auto instantiate(type input_schema, operator_control_plane& ctrl) const
+    -> caf::expected<std::unique_ptr<printer_instance>> override {
+    TENZIR_ASSERT_CHEAP(inner_);
+    return inner_->instantiate(std::move(input_schema), ctrl);
+  }
+
+  auto allows_joining() const -> bool override {
+    TENZIR_ASSERT_CHEAP(inner_);
+    return inner_->allows_joining();
```

Comment:
Since we don't print schema information we can always join the output across multiple schemas.

Reply:
Making this return `true` unconditionally will cause an assertion failure in `xsv_printer`.

Reply:
`allows_joining()` causes the `write` operator to choose a different code path, which `xsv_printer` doesn't support.

Reply:
Does it make sense to generally allow the `xsv_printer` to join between schemas when not printing a header?

Reply:
That would mean some refactoring in the `xsv_printer`. I'll write an issue about that, and follow up separately.